tests/gold_tests/logging/sigusr2.test.py

```diff
@@ -0,0 +1,232 @@
+rotate_manager_log = tr1.Processes.Process("rotate_manager_log", "mv {} {}".format(
+    diags_test.manager_log, diags_test.rotated_manager_log))
+
+# Configure the signaling of SIGUSR2 to traffic_manaager.
+tr1.Processes.Default.Command = "pkill -n -SIGUSR2 traffic_manager"
+tr1.Processes.Default.Return = 0
+tr1.Processes.Default.Ready = When.FileExists(diags_test.diags_log)
+
+# Configure process order.
+tr1.Processes.Default.StartBefore(rotate_diags_log)
+rotate_diags_log.StartBefore(rotate_manager_log)
+rotate_manager_log.StartBefore(diags_test.ts)
+tr1.StillRunningAfter = diags_test.ts
+tr1.StillRunningAfter = diags_test.server
+
+# manager.log should have been rotated. Check for the expected content in the
+# old file and the newly created file.
+diags_test.ts.Disk.manager_log_old.Content += Testers.ContainsExpression(
+        "received SIGUSR2, rotating the logs",
+        "manager.log_old should explain that SIGUSR2 was passed to it")
+
+diags_test.ts.Disk.manager_log.Content += Testers.ContainsExpression(
+        "opened.*manager.log",
+        "The new manager.log should indicate the newly opened manager.log")
+
+# diags.log should have been rotated. The old one had the reference to traffic
+# server running, this new one shouldn't. But it should indicate that the new
+# diags.log was opened.
+diags_test.ts.Disk.diags_log.Content += Testers.ExcludesExpression(
+        "traffic server running",
+        "The new diags.log should not reference the running traffic server")
+
+diags_test.ts.Disk.diags_log.Content += Testers.ContainsExpression(
+        "opened.*diags.log",
+        "The new diags.log should indicate the newly opened diags.log")
+
+#
+# Test 2: Verify SIGUSR2 isn't needed for rotated configured logs.
+#
+tr2 = Test.AddTestRun("Verify yaml.log logs can be rotated")
+configured_test = Sigusr2Test()
+
+first_curl = tr2.Processes.Process(
+        "first_curl",
+        'curl "http://127.0.0.1:{0}/first" --verbose'.format(configured_test.ts.Variables.port))
+first_curl_ready = tr2.Processes.Process("first_curl_ready", 'sleep 30')
+# In the autest enironment, it can take more than 10 seconds for the log file to be created.
+first_curl_ready.StartupTimeout = 30
+first_curl_ready.Ready = When.FileExists(configured_test.configured_log)
+
+rotate_log = tr2.Processes.Process("rotate_log_file", "mv {} {}".format(
+    configured_test.configured_log, configured_test.rotated_configured_log))
+
+second_curl = tr2.Processes.Process(
+        "second_curl",
+        'curl "http://127.0.0.1:{0}/second" --verbose'.format(configured_test.ts.Variables.port))
+
+second_curl_ready = tr2.Processes.Process("second_curl_ready", 'sleep 30')
+# In the autest enironment, it can take more than 10 seconds for the log file to be created.
+second_curl_ready.StartupTimeout = 30
+second_curl_ready.Ready = When.FileModified(configured_test.rotated_configured_log)
+
+send_pkill = tr2.Processes.Process("Send SIGUSR2", "pkill -n -SIGUSR2 traffic_manager")
```

Comment:
I'm probably being paranoid, but on the CI machines, can two Dockers be running Au tests be in the same VM (or directly running on the same physical host)?  And could that result in pkill seeing "the last run" traffic_manager being traffic_manager run in a different container?  I wonder if there's some method like tr2.pid() that could give the PID to use with just plain ole kill.

Reply:
It doesn't look like Au test has a built-in mechanism to get the PID of a running process.

Reply:
Here's an alternative (but ugly) approach that avoids using the pkill command:  https://github.com/ywkaras/trafficserver/commit/4e3d38413136f789ed6c568101a282ac1086313c

Reply:
aren't the processes isolated through docker/process namespacing? 

and instead of resorting to shell ps/grep combos, why not use something more native, like psutil? (other tests use psutil)

Reply:
Good thoughts Walt and Randall!

I implemented this using a script utilizing psutil as @randall suggested which takes a process identifier to uniquely identify the process as @ywkaras gave an example of on his branch.